File: sdk_files/v33/javascript/oauth_authorization.js
import { handleResponse } from './errors.js';

export class OauthAuthorization {
  constructor(ctx) { this._ctx = ctx; }

  async authorize({ responseType, clientId, redirectUri, state, codeChallenge, codeChallengeMethod } = {}) {
    let path = `/oauth/authorize`;
    const query = new URLSearchParams();
    if (responseType !== undefined) query.set('response_type', responseType);
    if (clientId !== undefined) query.set('client_id', clientId);
    if (redirectUri !== undefined) query.set('redirect_uri', redirectUri);
    if (state !== undefined) query.set('state', state);
    if (codeChallenge !== undefined) query.set('code_challenge', codeChallenge);
    if (codeChallengeMethod !== undefined) query.set('code_challenge_method', codeChallengeMethod);
    const qs = query.toString();
    if (qs) path += '?' + qs;
    const resp = await fetch(this._ctx.baseUrl + path, { method: 'GET' });
    return handleResponse(resp);
  }

  async getConsent(sessionId) {
    const resp = await fetch(this._ctx.baseUrl + `/oauth/consent/${encodeURIComponent(sessionId)}`, { method: 'GET' });
    return handleResponse(resp);
  }

  async postApproveConsent(sessionId) {
    const resp = await fetch(this._ctx.baseUrl + `/oauth/consent/${encodeURIComponent(sessionId)}/approve`, { method: 'POST' });
    return handleResponse(resp);
  }

  async postDenyConsent(sessionId) {
    const resp = await fetch(this._ctx.baseUrl + `/oauth/consent/${encodeURIComponent(sessionId)}/deny`, { method: 'POST' });
    return handleResponse(resp);
  }
}

File: sdk/v1.0.69/python/mediaviz_sdk/o_auth_authorization.py
from __future__ import annotations
from typing import Any
from urllib.parse import quote, urlencode
import httpx

from .errors import handle_response


class OAuthAuthorization:
    def __init__(self, ctx) -> None:
        self._ctx = ctx

    def authorize(
        self,
        response_type: str | None = None,
        client_id: str | None = None,
        redirect_uri: str | None = None,
        state: str | None = None,
        code_challenge: str | None = None,
        code_challenge_method: str | None = None,
    ) -> dict[str, Any]:
        path = '/oauth/authorize'
        _q: dict[str, Any] = {}
        if response_type is not None:
            _q['response_type'] = response_type
        if client_id is not None:
            _q['client_id'] = client_id
        if redirect_uri is not None:
            _q['redirect_uri'] = redirect_uri
        if state is not None:
            _q['state'] = state
        if code_challenge is not None:
            _q['code_challenge'] = code_challenge
        if code_challenge_method is not None:
            _q['code_challenge_method'] = code_challenge_method
        if _q:
            path += '?' + urlencode(_q, doseq=True)
        with httpx.Client() as _client:
            _resp = _client.request('GET', self._ctx.base_url + path)
        return handle_response(_resp.text, _resp.status_code, dict(_resp.headers))

    def get_consent(self, session_id: str) -> dict[str, Any]:
        path = '/oauth/consent/' + quote(str(session_id), safe='')
        with httpx.Client() as _client:
            _resp = _client.request('GET', self._ctx.base_url + path)
        return handle_response(_resp.text, _resp.status_code, dict(_resp.headers))

    def post_approve_consent(self, session_id: str, restartUrl: Any) -> dict[str, Any]:
        path = '/oauth/consent/' + quote(str(session_id), safe='') + '/approve'
        body = {
            'restart_url': restartUrl,
        }
        with httpx.Client() as _client:
            _resp = _client.request('POST', self._ctx.base_url + path, data=body)
        return handle_response(_resp.text, _resp.status_code, dict(_resp.headers))

    def post_deny_consent(self, session_id: str, restartUrl: Any) -> dict[str, Any]:
        path = '/oauth/consent/' + quote(str(session_id), safe='') + '/deny'
        body = {
            'restart_url': restartUrl,
        }
        with httpx.Client() as _client:
            _resp = _client.request('POST', self._ctx.base_url + path, data=body)
        return handle_response(_resp.text, _resp.status_code, dict(_resp.headers))

    def get_switch_user(self, session_id: str, restart_url: str | None = None) -> dict[str, Any]:
        path = '/oauth/consent/' + quote(str(session_id), safe='') + '/switch-user'
        _q: dict[str, Any] = {}
        if restart_url is not None:
            _q['restart_url'] = restart_url
        if _q:
            path += '?' + urlencode(_q, doseq=True)
        with httpx.Client() as _client:
            _resp = _client.request('GET', self._ctx.base_url + path)
        return handle_response(_resp.text, _resp.status_code, dict(_resp.headers))
